Expand footer lang switch to avoid cutting off long word (fix #15609)

wen-2018 opened the pull request:
## One-line summary

This PR expands footer lang switch to avoid cutting off long word.

## Significant changes and points to review

- Footer lang switch shouldn't be cutting off any language

## Issue / Bugzilla link

https://github.com/mozilla/bedrock/issues/15609

## Testing

In order to avoid cutting off any language, the select is quite long and it looks kind of broken when no hovered. 
<img width="343" alt="normal" src="https://github.com/user-attachments/assets/133649ad-e7c3-46df-8724-2d3db0bb1a18">
<img width="378" alt="hover" src="https://github.com/user-attachments/assets/7e7943ed-deca-4a03-8ce3-27b20b15ecff">

Same approach worked on the older footer cause the district background color made it clear that the triangle is part of the select dropdown. Might wait when the designer is back for some design tweaks on this.
<img width="272" alt="old" src="https://github.com/user-attachments/assets/9dd58173-7ee0-4c8e-b52b-acb1691b59e2">


Tess's take on this:

- default border for the select
- on hover, inverse the colors: black text on white background

janbrasna commented on media/css/m24/components/footer-refresh.scss:
```diff
@@ -316,21 +295,57 @@ $max-footer-content-width: $content-max;
+        }
+
+        &::after {
+            @include bidi(((background-position, calc(100% - 8px) 0.75em, 8px 0.75em),));
```
This may be easier as:
```suggestion
            @include bidi(((background-position, right 8px center, 8px center),));
```

wen-2018 replied:
Great suggestion!

stephaniehobson commented on media/css/m24/components/footer-refresh.scss:
```diff
@@ -316,21 +317,25 @@ $max-footer-content-width: $content-max;
         &:focus-visible {
-            border-color: $m24-color-mid-gray;
+            color: $m24-color-white;
+            background: $m24-color-alt-black url('/media/img/icons/m24-small/up-caret.svg') no-repeat;
```
**Nit:** (optional) Let's have the caret point down all the time to match the behaviour on the newsletter select boxes.

wen-2018 replied:
Changing the caret to point up was specifically requested by Tess. This seems like a common behavior for a select. Should we update the newsletter select boxes to point up when open instead?

stephaniehobson replied:
It should at least point down until it is open, but I will discuss it with Tess.

stephaniehobson reviewed (changes requested):
This looks great! Seeing in on the page Tess and I would like to right align the social media links above it please:

![Screenshot 2024-12-12 at 13 50 56](https://github.com/user-attachments/assets/f948882f-ae70-4261-8fdc-9a4d2b99dde5)

stephaniehobson commented:
Needs a rebsase.

stephaniehobson commented:
You can decide if you want to include the select changes in the rebase or not. I said it was a nit and will not block on it.

wen-2018 commented:
@stephaniehobson I have rebased the PR and removed the up-caret change on hover. I can open a new PR if we decided to change the caret to point up when select is up. This PR at least will fix the broken styles

stephaniehobson reviewed (approved):
R+ 🥕 

Sorry for the back and fourth on the caret.
